### DISCUSSION TITLE
APN Autoconf in Android Phones for Private LTE Nets

### DISCUSSION THREAD

**Discussion by jpad-freedomfi** [magma/magma#7476] [Q&A]
Hi All, 

Curious if others have solved this issue. In some of our customer deployments we see android phones with no APN configured unable to get data services in a magma network using a shared PLMN. This requires users to manually configure the APN in the phone before data services work. 

In some markets, this manual APN configuration is too complex for many/most users.

I understand android supports carrier config settings and Apple has carrier bundles, but for smaller operators these can take a long time or be impossible to get. 

Have other had/solved this problem?

**Comment by pramchan** [comment]
Not sure why smaller operators can not provide you unless the devices they support for access are non-standard or not-properly documented in which case device manufacturer should be able to help working with the smaller operators.

**Comment by edaspb** [comment]
Manual APN config is a headache. The alternative option to manual configuration could be carrier app. https://source.android.com/devices/tech/config/carrier
OpenSource example for VoLTE https://github.com/herlesupreeth/CoIMS_Wiki

**Comment by jpad-freedomfi** [comment]
one update to share, Claudio on the FF team figured out that Android Carrier Config contains an entry for MCC 902 MNC 12 (not currently in use best I can tell) that has default APN of "internet" and operator name of "internet". When we program a SIM with IMSI using this PLMN, Android phones auto-populate APN = internet. 

Could be handy. 

Also, question, does anyone know if Magma SGS SMS API support binary data as payload? Looks like you can send APN configuration as binary blob via SMS to android and iOS phones. Curious if we can use the SMS API to deliver this data.

**Reply by jpad-freedomfi** [reply]
@sudhikan do you know about the SMS API support for binary blobs?

**Reply by sudhikan** [reply]
@shaddi :-) ?

**Reply by shaddi** [reply]
Yes, we should support that on the AGW. IIRC, the REST API on the orc8r, which is responsible for encoding SMS PDUs, may not be able to encode/decode these, however, but that should be a straightforward (if tedious) extension to the orc8r service and `sms_ll` library.

The 90212 PLMIND find seems super useful! MCC 902 seems to be mentioned [here](https://www.itu.int/md/T09-SG02-C-0130/_page.print), which of course is behind a paywall. Would be curious to know what that document says if someone in the community has access.
